var/spack/repos/builtin/packages/cp2k/package.py

```diff
@@ -763,3 +824,104 @@ def check(self):
         with spack.util.environment.set_env(CP2K_DATA_DIR=data_dir, PWD=self.build_directory):
             with working_dir(self.build_directory):
                 make("test", *self.build_targets)
+
+
+class CMakeBuilder(spack.build_systems.cmake.CMakeBuilder):
+    """Use the new cmake build system to build cp2k. It is the default when
+    building the master branch of cp2k."""
+
+    def cmake_args(self):
+        spec = self.spec
+
+        if "+cuda" in spec and len(spec.variants["cuda_arch"].value) > 1:
+            raise InstallError("CP2K supports only one cuda_arch at a time")
+
+        if "+rocm" in spec and len(spec.variants["amdgpu_target"].value) > 1:
+            raise InstallError("CP2K supports only one amdgpu_arch at a time")
+
+        args = []
+
+        if spec.satisfies("+cuda") or spec.satisfies("+rocm"):
+            if spec.satisfies("cuda_target"):
+                args += [self.define("CP2K_WITH_GPU", gpu_map[spec.variants["cuda_arch"].value])]
+
+            if spec.satisfies("amdgpu_target"):
+                args += [
+                    self.define("CP2K_WITH_GPU", gpu_map[spec.variants["amdgpu_target"].value])
```

Comment:
I think `value` is an array with one element at this point:
```suggestion
                args += [self.define("CP2K_WITH_GPU", gpu_map[spec.variants["cuda_arch"].value[0])]

            if spec.satisfies("amdgpu_target"):
                args += [
                    self.define("CP2K_WITH_GPU", gpu_map[spec.variants["amdgpu_target"].value[0])
```